### PR TITLE
Conceptual data model

### PR DESCRIPTION
**Pull request by jaspersprengers**

### CODE REVIEW

File: model.md [hunk 1]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
+* Country. At present littil is only active in the Netherlands, but this can change.
+* Date of last login/activity. Maintained by the platform, not part of the profile.
+* Preferences: for the time being this will be free text. Possibly later you can choose from a set of modules. 
+* Availability: which days of the week are you in principle available?  
+
+## The school profile
+* unique identifier, generated by the platform
+* Name of the school
+* Address of the school. Full address is optional. Postal code is mandatory.
+* Name of contact person
```
Comment (ramondebeijer) [inline model.md:22]: split in firstname / surname?

File: model.md [hunk 2]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
-  * What is their area of expertise, with regard to the choice of classes they might provide?
-  * What weekdays are they available?
-  * How far are they willing to travel?
+* unique identifier, generated by the platform. Preferably a UUID, not a DB sequence number.
+* First name
+* Surname
+* Email (must be unique)
+* Postal code, at minimum the digits, not the letters. This is all that is needed for a geographical match.
+* Country. At present littil is only active in the Netherlands, but this can change.
+* Date of last login/activity. Maintained by the platform, not part of the profile.
```
Comment (pschildkamp) [inline model.md:14]: Why? Doesn't solutions like Auth0 (which we are looking into if it's a suitable solution for us) keep track of these kind of data? If we don't want so show others the last activity I don't see a reason to keep track on in within the platform itself.

**Review by ramondebeijer — APPROVED** [review]

File: model.md [hunk 3]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
-    * Place of residence, possibly only postal code. This is necessary to make a geographical match.
-    * Country. At present littil is only active in the Netherlands, but this can change
-    * Non-essential personal data would be date of birth, gender, telephone, full postal address. It is debatable whether we need at all. The less, the better
-    * Nice to have and without obvious privacy constraints would be a link to a public profile, like a personal website or LinkedIn. A few lines of bio would also be good. 
-* **Teaching skills, preferences and availability**
-  * How experienced is the teacher in giving classes in general? Have they given similar classes before?
-  * What is their area of expertise, with regard to the choice of classes they might provide?
-  * What weekdays are they available?
-  * How far are they willing to travel?
+* unique identifier, generated by the platform. Preferably a UUID, not a DB sequence number.
```
Comment (pschildkamp) [inline model.md:8]: Do we need to think about the version of the UUID? From what I've read Mysql support version 1 out of the box. Default of java (randomUUID()) is version 4. Anyone experience with Mysql and UUID?

File: model.md [hunk 4]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
+* First name
+* Surname
+* Email (must be unique)
+* Postal code, at minimum the digits, not the letters. This is all that is needed for a geographical match.
+* Country. At present littil is only active in the Netherlands, but this can change.
+* Date of last login/activity. Maintained by the platform, not part of the profile.
+* Preferences: for the time being this will be free text. Possibly later you can choose from a set of modules.
```
Comment (pschildkamp) [inline model.md:15]: Maybe we should consider skipping this property if we don't have a use case at the moment. We can always add them in a later stage.

**Review by pschildkamp — COMMENTED** [review]
Just a general thought. If we want to implement a third party authentication system don't we need some kind of user <> teacher and user <> school mapping entity table?

File: model.md (hunk 3 above)
Comment (Stefanic) [reply model.md:8]: Is this an extra identifier apart from a primary key field? Don't know why it explicitly needs to not be a DB sequence number, but if you want this UUID to be in the URL or something a v1 might be random enough.

**Review by Stefanic — APPROVED** [review]

File: model.md (hunk 1 above)
Comment (jaspersprengers) [reply model.md:22]: Yes

File: model.md [hunk 5]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
+* Postal code, at minimum the digits, not the letters. This is all that is needed for a geographical match.
+* Country. At present littil is only active in the Netherlands, but this can change.
+* Date of last login/activity. Maintained by the platform, not part of the profile.
+* Preferences: for the time being this will be free text. Possibly later you can choose from a set of modules. 
+* Availability: which days of the week are you in principle available?  
+
+## The school profile
+* unique identifier, generated by the platform
+* Name of the school
+* Address of the school. Full address is optional. Postal code is mandatory.
```
[the remark this answers is not in the document]
Comment (jaspersprengers) [reply model.md:21]: Yes

File: model.md [hunk 6]
```diff
@@ -5,19 +5,26 @@ Consider the basic scenario of a school looking for a list of suitable and avail
...
-    * Non-essential personal data would be date of birth, gender, telephone, full postal address. It is debatable whether we need at all. The less, the better
-    * Nice to have and without obvious privacy constraints would be a link to a public profile, like a personal website or LinkedIn. A few lines of bio would also be good. 
-* **Teaching skills, preferences and availability**
-  * How experienced is the teacher in giving classes in general? Have they given similar classes before?
-  * What is their area of expertise, with regard to the choice of classes they might provide?
-  * What weekdays are they available?
-  * How far are they willing to travel?
+* unique identifier, generated by the platform. Preferably a UUID, not a DB sequence number.
+* First name
+* Surname
```
[the remark this answers is not in the document]
Comment (jaspersprengers) [reply model.md:10]: Yes, I propose not to make it too difficult with a separate field for these 'voorvoegsels'

File: model.md (hunk 3 above)
Comment (jaspersprengers) [reply model.md:8]: Not being tied to the DB's mechanism of auto-key generation makes it more portable, I think.

File: model.md (hunk 4 above)
Comment (jaspersprengers) [reply model.md:15]: which one?

File: model.md (hunk 3 above)
Comment (Stefanic) [reply model.md:8]: True, then we should also generate UUIDs in code I guess.

**Pull request merged**